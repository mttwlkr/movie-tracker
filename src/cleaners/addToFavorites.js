const url = process.env.REACT_APP_DATABASE_URL;

export const addToFavorites = async (movie) => {
  try {
    const response = await fetch(`${url}/api/v1/favorites`, {
      method: 'POST',
      body: JSON.stringify(movie),
      headers: {
        'content-type': 'application/json'
      }
    });
    const favoriteData = await response.json();
   
    return favoriteData;
  } catch (error) {
    throw error;
  }
};