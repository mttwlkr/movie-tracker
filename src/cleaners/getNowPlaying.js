const apiKey = process.env.REACT_APP_MOVIE_DB_KEY;

export const getNowPlaying = async () => {
  const rootUrl = 'https://api.themoviedb.org/3/movie/now_playing?page=1';
  try {
    const response = await fetch(
      `${rootUrl}&language=en-US&api_key=${apiKey}&language=en-US`);
    const nowPlayingData = await response.json();
    const cleanedMovies = cleanMovies(nowPlayingData);
    return cleanedMovies;
  } catch (error) {
    throw error;
  }
};

export const cleanMovies = (movieData) => {
  return movieData.results.map(movie => ({
    movie_id: movie.id, 
    title: movie.title, 
    poster_path: movie.poster_path, 
    release_date: movie.release_date, 
    vote_average: movie.vote_average, 
    overview: movie.overview  
  }));
};