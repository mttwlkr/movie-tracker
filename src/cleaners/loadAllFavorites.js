const url = process.env.REACT_APP_DATABASE_URL;

export const loadAllFavorites = async (userId) => {

  try {
    const response = await fetch(`${url}/api/v1/users/${userId}/favorites`);

    if (response.status === 404) {      
      return {};
    }
    
    return await response.json();
  } catch (error) {
    throw error;
  }
};