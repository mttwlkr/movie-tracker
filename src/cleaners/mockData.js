/* eslint-disable */

export const mocklogInData = { 
  id: 1,
  name: "Taylor",
  message: "Retrieved ONE User" 
};

export const mockNowPlaying = {
  results: [
    {
      "adult" : "false",
      "backdrop_path" : "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
      "genre_ids" : "[28, 12, 14, 878]",
      "id" : "284054",
      "original_language" : "en",
      "original_title" : "Black Panther",
      "overview" : "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "popularity" : "249.862248",
      "poster_path" : "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "release_date" : "2018-02-13",
      "title" : "Black Panther",
      "video" : "false",
      "vote_average" : "7.4",
      "vote_count" : "3582"
    },
    {
      "adult" : "false",
      "backdrop_path" : "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
      "genre_ids" : "[28, 12, 14, 878]",
      "id" : "284054",
      "original_language" : "en",
      "original_title" : "Black Panther",
      "overview" : "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "popularity" : "249.862248",
      "poster_path" : "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "release_date" : "2018-02-13",
      "title" : "Black Panther",
      "video" : "false",
      "vote_average" : "7.4",
      "vote_count" : "3582"
    },
    {
      "adult" : "false",
      "backdrop_path" : "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
      "genre_ids" : "[28, 12, 14, 878]",
      "id" : "284054",
      "original_language" : "en",
      "original_title" : "Black Panther",
      "overview" : "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "popularity" : "249.862248",
      "poster_path" : "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "release_date" : "2018-02-13",
      "title" : "Black Panther",
      "video" : "false",
      "vote_average" : "7.4",
      "vote_count" : "3582"
    }, 
    {
      "adult" : "false",
      "backdrop_path" : "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
      "genre_ids" : "[28, 12, 14, 878]",
      "id" : "284054",
      "original_language" : "en",
      "original_title" : "Black Panther",
      "overview" : "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "popularity" : "249.862248",
      "poster_path" : "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "release_date" : "2018-02-13",
      "title" : "Black Panther",
      "video" : "false",
      "vote_average" : "7.4",
      "vote_count" : "3582"
    },
    {
      "adult" : "false",
      "backdrop_path" : "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
      "genre_ids" : "[28, 12, 14, 878]",
      "id" : "284054",
      "original_language" : "en",
      "original_title" : "Black Panther",
      "overview" : "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "popularity" : "249.862248",
      "poster_path" : "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "release_date" : "2018-02-13",
      "title" : "Black Panther",
      "video" : "false",
      "vote_average" : "7.4",
      "vote_count" : "3582"
    }       
  ]
}

export const mockMovieProp = [{
  "adult" : "false",
  "backdrop_path" : "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
  "genre_ids" : "[28, 12, 14, 878]",
  "id" : "284054",
  "original_language" : "en",
  "original_title" : "Black Panther",
  "overview" : "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
  "popularity" : "249.862248",
  "poster_path" : "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
  "release_date" : "2018-02-13",
  "title" : "Black Panther",
  "video" : "false",
  "vote_average" : "7.4",
  "vote_count" : "3582"
}];    

export const mockCleanNowPlaying = [{
   "movie_id": "284054",
   "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
   "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
    "release_date": "2018-02-13",
   "title": "Black Panther",
   "vote_average": "7.4",
}]

export const mockCleanMovie = [{
   "movie_id": "284054",
   "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
   "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
    "release_date": "2018-02-13",
   "title": "Black Panther",
   "vote_average": "7.4",
}]

export const mockMovie = {
  results: [
    {
      "adult" : "false",
      "backdrop_path" : "/ut3n6iLMF79vDqyKFU6PeojZZpQ.jpg",
      "genre_ids" : "[28, 12, 14, 878]",
      "id" : "284054",
      "original_language" : "en",
      "original_title" : "Black Panther",
      "overview" : "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
      "popularity" : "249.862248",
      "poster_path" : "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
      "release_date" : "2018-02-13",
      "title" : "Black Panther",
      "video" : "false",
      "vote_average" : "7.4",
      "vote_count" : "3582"
    }
  ]
};

export let mockUser = {
  name: 'mud',
  id: 4
}

export const mockLoadAllFavoritesMovie = [{
   "movie_id": "284054",
   "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
   "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
    "release_date": "2018-02-13",
   "title": "Black Panther",
   "vote_average": "7.4",
}]

export const mockAddFavoriteMovie = {"0": {"movie_id": "284054", "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.", "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg", "release_date": "2018-02-13", "title": "Black Panther", "vote_average": "7.4"}, "user_id": 4}; 