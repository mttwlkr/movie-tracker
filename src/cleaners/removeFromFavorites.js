const url = process.env.REACT_APP_DATABASE_URL;

export const removeFromFavorites = async (user_id, movie_id) => {
  const removeInfo = JSON.stringify({ user_id, movie_id });
  
  try {
    const response = await fetch(`${url}/api/v1/favorites/${removeInfo}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      }
    });
    const removedData = await response.json();
    
    return removedData;
  } catch (error) {
    throw error;
  }
};