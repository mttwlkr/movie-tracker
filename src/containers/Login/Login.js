import React, { Component } from 'react';
import { withRouter, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import './Login.css';
import { logInUser, addAllFavorites } from '../../actions';
import { loadAllFavorites } from '../../cleaners/loadAllFavorites.js';
import PropTypes from 'prop-types';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      error: false
    };
  }

  submitEmail = (event) => {
    event.preventDefault();
    const userInfo = { email: this.state.email, password: this.state.password };
    this.logIn(userInfo);
  }

  logIn = async (userInfo) => {
    try {
      const response = await fetch('/api/users', {
        method: "POST",
        body: JSON.stringify(userInfo),
        headers: {
          'content-type': 'application/json'
        }
      });
      const logInData = await response.json();
      
      this.redirectUser(logInData.data.id, logInData.data.name);
      this.showFavorites(logInData.data.id);
    } catch (error){
      this.setState({error: true});
    }
  }

  showFavorites = async (userId) => {
    const allFavorites = await loadAllFavorites(userId);

    this.props.addAllFavorites(allFavorites);
  }

  redirectUser = (id, name) => {
    this.props.handleSubmit(id, name);
    const path = "/";
    this.props.history.push(path);
  }

  render() {
    return (
      <div>
        <form onSubmit={this.submitEmail}>
          <input 
            type='text' 
            id='email' 
            value={this.state.email} 
            placeholder='Email'
            onChange={(event) => this.setState({ 
              email: event.target.value })}
          />
          <input 
            type='password' 
            id='password' 
            value={this.state.password} 
            placeholder='Password'
            onChange={(event) => this.setState({ 
              password: event.target.value })}
          />
          <button className='submit'>Submit</button>
        </form>
        { this.state.error &&
            <section>
              <p>Email and password do not match, please try again.</p>
              <NavLink to="/signup">Sign Up</NavLink>
            </section>
        }
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    user: state.user
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    handleSubmit: (id, name) => (dispatch(logInUser(id, name))),
    addAllFavorites: (movies) => (dispatch(addAllFavorites(movies)))
  };
};  

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login));

Login.propTypes = {
  history: PropTypes.array,
  handleSubmit: PropTypes.func,
  addAllFavorites: PropTypes.func
};