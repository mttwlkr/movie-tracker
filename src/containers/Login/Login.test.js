import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';
import { loadAllFavorites } from '../../cleaners/loadAllFavorites.js';
import { mocklogInData } from '../../cleaners/mockData';

jest.mock('../../cleaners/loadAllFavorites');

describe('Login', () => {
  let wrapper, 
    mockHandleSubmit, 
    mockEvent, 
    mockHistory,
    mockAddAllFavorites;

  beforeEach(() => {
    mockHandleSubmit = jest.fn();
    mockEvent = {
      preventDefault: jest.fn()
    };

    mockHistory = { push: jest.fn() };
    mockAddAllFavorites = jest.fn();
    wrapper = shallow(<Login 
      handleSubmit={mockHandleSubmit}
      history={mockHistory} 
      addAllFavorites={mockAddAllFavorites} />
    );
  });

  it('should match the snapshot', ()=> {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke logIn with correct params on submitEmail', () => {
    const expected = {
      email: '',
      password: ''
    };

    wrapper.instance().logIn = jest.fn();
    const spy = jest.spyOn(wrapper.instance(), 'logIn');

    wrapper.instance().submitEmail(mockEvent);
    expect(spy).toHaveBeenCalledWith(expected);
  });

  it('should call redirectUser with correct params', async () => {
    const mockData = {
      email: 'tman2272@aol.com',
      password: 'password'
    };

    wrapper.instance().redirectUser = jest.fn();
    const spy = jest.spyOn(wrapper.instance(), 'redirectUser');

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      ok: true,
      json: () => Promise.resolve(mocklogInData)
    }));

    await wrapper.instance().logIn(mockData);
    expect(spy).toHaveBeenCalledWith(1, 'Taylor');
  });

  it('should fetch favorites from db and put them into the store', async () => {
    
    await wrapper.instance().showFavorites();
    expect(mockAddAllFavorites).toHaveBeenCalled();
  });

  it('should loadAllFavorites when showFavorites is called', () => {
    const id ='666';

    wrapper.instance().showFavorites(id);
    expect(loadAllFavorites).toHaveBeenCalledWith(id);
  });

  it('should call handleSubmit when redirectUser is called', () => {
    const id = '122345';
    const name = 'jared';

    wrapper.instance().redirectUser(id, name);
    expect(mockHandleSubmit).toHaveBeenCalledWith(id, name);
  });

  it('should set state error to true in case of bad response', async () => {
    const mockData = {
      email: 'tman@gmail.com',
      password: 'blobs'
    };

    window.fetch = jest.fn().mockImplementation(() => Promise.reject({
      status: 500
    }));

    await wrapper.instance().logIn(mockData);
    expect(wrapper.state('error')).toEqual(true);
  });

  it('should set state of email and password when input changes', () => {
    const event1 = { target: { value: 'tman@gmail.com' } };

    wrapper.find('#email').simulate('change', event1);
    expect(wrapper.state('email')).toEqual('tman@gmail.com');

    const event2 = { target: { value: 'dayman' } };

    wrapper.find('#password').simulate('change', event2);
    expect(wrapper.state('password')).toEqual('dayman');
  });

  it('should map to the store correctly', () => {
    const mockStore = { user: {name: 'Jared'} };

    const mapped = mapStateToProps(mockStore);
    expect(mapped.user).toEqual(mockStore.user);
  });

  it('should dispatch to the store correctly', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.handleSubmit();
    expect(mockDispatch).toHaveBeenCalled();
  });

});