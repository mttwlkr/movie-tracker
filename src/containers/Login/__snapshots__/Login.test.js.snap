// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    addAllFavorites={[Function]}
    handleSubmit={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    id="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
          />
          <input
                    id="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <button
                    className="submit"
          >
                    Submit
          </button>
</form>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
/>,
            <input
              id="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <button
              className="submit"
>
              Submit
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        id="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="text"
                        value=""
            />
            <input
                        id="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
            <button
                        className="submit"
            >
                        Submit
            </button>
</form>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                id="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
/>,
              <input
                id="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
/>,
              <button
                className="submit"
>
                Submit
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
